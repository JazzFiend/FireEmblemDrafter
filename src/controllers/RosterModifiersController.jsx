import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import RosterModifiers from '../view/RosterModifiers';
import { selectDraftInProgress } from '../features/draft/DraftInProgressSlice';
import { selectRequiredRestricted, setRequired, setRestricted } from '../features/draft/RequiredRestrictedSlice';
import { selectRoster } from '../features/draft/RosterSlice';

export default function RosterModifiersController() {
  const draftInProgress = useSelector(selectDraftInProgress);
  const requiredRestrictedCharacters = useSelector(selectRequiredRestricted);
  const allCharacters = useSelector(selectRoster);
  const dispatch = useDispatch();

  function addDisables(disabledList, fullListKeyed) {
    const keyedWithDisabled = fullListKeyed.map((keyedElement) => {
      if (disabledList.includes(keyedElement.label)) {
        const disabledItem = {
          label: keyedElement.label,
          value: keyedElement.value,
          isDisabled: true,
        };
        return disabledItem;
      }
      return keyedElement;
    });
    return keyedWithDisabled;
  }

  function keyList(list) {
    return list.map((element, index) => ({
      value: index,
      label: element,
    }));
  }

  function populateDefaultValues(defaultStringList, keyedListWithDisables) {
    return defaultStringList.map(
      (itemString) => keyedListWithDisables.find(
        (requiredOption) => itemString === requiredOption.label,
      ),
    );
  }

  function showDraftOptions() {
    return (!draftInProgress && allCharacters.length > 0);
  }

  const handleUpdateRequiredUnits = (requiredUnits) => {
    dispatch(setRequired(requiredUnits));
  };

  const handleUpdateRestrictedUnits = (requiredUnits) => {
    dispatch(setRestricted(requiredUnits));
  };

  const fullListKeyed = keyList(allCharacters);
  const restrictedOptionsWithDisables = addDisables(requiredRestrictedCharacters.requiredUnits, fullListKeyed);
  const requiredOptionsWithDisables = addDisables(requiredRestrictedCharacters.restrictedUnits, fullListKeyed);
  const requiredDefaults = populateDefaultValues(
    requiredRestrictedCharacters.requiredUnits,
    requiredOptionsWithDisables,
  );
  const restrictedDefaults = populateDefaultValues(
    requiredRestrictedCharacters.restrictedUnits,
    restrictedOptionsWithDisables,
  );

  return (
    <div>
      <RosterModifiers
        testId="required"
        placeholderText="Required Units"
        defaultValue={requiredDefaults}
        selected={requiredRestrictedCharacters.requiredUnits}
        optionsWithDisables={requiredOptionsWithDisables}
        onChangeHandler={handleUpdateRequiredUnits}
        showDraftOptions={showDraftOptions(allCharacters)}
        showDraftOptionsLists={draftInProgress}
      />

      <RosterModifiers
        testId="restricted"
        placeholderText="Restricted Units"
        defaultValue={restrictedDefaults}
        selected={requiredRestrictedCharacters.restrictedUnits}
        optionsWithDisables={restrictedOptionsWithDisables}
        onChangeHandler={handleUpdateRestrictedUnits}
        showDraftOptions={showDraftOptions(allCharacters)}
        showDraftOptionsLists={draftInProgress}
      />
    </div>
  );
}
