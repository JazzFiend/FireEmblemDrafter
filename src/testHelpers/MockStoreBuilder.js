import configureStore from 'redux-mock-store';

export default class MockStoreBuilder {
  constructor() {
    this.draftInProgress = false;
    this.requiredUnits = [];
    this.restrictedUnits = [];
    this.roster = [];
    this.teamSize = 0;
  }

  withDraftStatus(draftStatus) {
    this.draftInProgress = draftStatus;
    return this;
  }

  withRequiredUnits(requiredUnits) {
    this.requiredUnits = requiredUnits;
    return this;
  }

  withRestrictedUnits(restrictedUnits) {
    this.restrictedUnits = restrictedUnits;
    return this;
  }

  withRoster(roster) {
    this.roster = roster;
    return this;
  }

  withTeamSize(teamSize) {
    this.teamSize = teamSize;
    return this;
  }

  build() {
    const mockStore = configureStore([]);
    return mockStore({
      draftInProgress: {
        value: this.draftInProgress,
      },
      requiredRestricted: {
        value: {
          requiredUnits: this.requiredUnits,
          restrictedUnits: this.restrictedUnits,
        },
      },
      roster: {
        value: this.roster,
      },
      teamSize: {
        value: this.teamSize,
      },
    });
  }
}
