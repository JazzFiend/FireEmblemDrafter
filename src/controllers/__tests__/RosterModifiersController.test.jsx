import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import RosterModifiersController from '../RosterModifiersController';
import MockStoreBuilder from '../../testHelpers/MockStoreBuilder';

afterEach(cleanup);

test('should render nothing when draft has not started and no characters are loaded', () => {
  const store = new MockStoreBuilder().build();
  const { asFragment } = render(
    <Provider store={store}>
      <RosterModifiersController />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should render unit dropdowns when draft has not started and characters are loaded', () => {
  const store = new MockStoreBuilder().withRoster(['u', 'v', 'w', 'x', 'y', 'z']).build();
  const { asFragment } = render(
    <Provider store={store}>
      <RosterModifiersController />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should not render when a set of characters is not loaded', () => {
  const store = new MockStoreBuilder().withDraftStatus(false).build();
  const { asFragment } = render(
    <Provider store={store}>
      <RosterModifiersController
        allCharacters={[]}
      />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should render when a set of characters is loaded', () => {
  const store = new MockStoreBuilder().withDraftStatus(false).withRoster(['u', 'v', 'w', 'x', 'y', 'z']).build();
  const { asFragment } = render(
    <Provider store={store}>
      <RosterModifiersController />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should hide dropdown and show restricted units when draft has started', () => {
  const restrictedCharacters = ['y', 'z'];
  const allCharacters = ['u', 'v', 'w', 'x', 'y', 'z'];
  const store = new MockStoreBuilder().withDraftStatus(true).withRestrictedUnits(restrictedCharacters).build();

  const { asFragment } = render(
    <Provider store={store}>
      <RosterModifiersController
        allCharacters={allCharacters}
      />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should hide dropdown and show required units when draft has started', () => {
  const requiredCharacters = ['y', 'z'];
  const allCharacters = ['u', 'v', 'w', 'x', 'y', 'z'];
  const store = new MockStoreBuilder().withDraftStatus(true).withRequiredUnits(requiredCharacters).build();

  const { asFragment } = render(
    <Provider store={store}>
      <RosterModifiersController
        allCharacters={allCharacters}
      />
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('should run restrictedUnitSelector function when clicked', () => {
  const store = new MockStoreBuilder().withRoster(['u', 'v', 'w', 'x', 'y', 'z']).build();

  const { container } = render(
    <Provider store={store}>
      <RosterModifiersController />
    </Provider>,
  );
  userEvent.click(container.querySelector('.restricted-selector__control'));
  userEvent.click(container.querySelectorAll('.restricted-selector__option')[0]);

  const storeActions = store.getActions();
  expect(storeActions.length).toEqual(1);
  expect(storeActions[0].type).toEqual('requiredRestricted/setRestricted');
  expect(storeActions[0].payload).toEqual(['u']);
});

test('should run requiredUnitSelector function when clicked', () => {
  const store = new MockStoreBuilder().withRoster(['u', 'v', 'w', 'x', 'y', 'z']).build();

  const { container } = render(
    <Provider store={store}>
      <RosterModifiersController />
    </Provider>,
  );
  userEvent.click(container.querySelector('.required-selector__control'));
  userEvent.click(container.querySelectorAll('.required-selector__option')[0]);

  const storeActions = store.getActions();
  expect(storeActions.length).toEqual(1);
  expect(storeActions[0].type).toEqual('requiredRestricted/setRequired');
  expect(storeActions[0].payload).toEqual(['u']);
});

test('options should disappear and become disabled in the required menu', () => {
  const store = new MockStoreBuilder()
    .withDraftStatus(false)
    .withRequiredUnits(['w'])
    .withRestrictedUnits(['u', 'v'])
    .withRoster(['u', 'v', 'w', 'x', 'y', 'z'])
    .build();

  const { asFragment, container } = render(
    <Provider store={store}>
      <RosterModifiersController />
    </Provider>,
  );
  userEvent.click(container.querySelector('.required-selector__control'));

  expect(asFragment()).toMatchSnapshot();
});
