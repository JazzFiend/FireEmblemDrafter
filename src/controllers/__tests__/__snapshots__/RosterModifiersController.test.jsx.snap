// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options should disappear and become disabled in the required menu 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-testid="required-drop-down"
      >
        <div
          class="required-selector css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          >
            <span
              id="aria-selection"
            />
            <span
              id="aria-context"
            >
              option u focused disabled, 1 of 6. 5 results available. Use Up and Down to choose options, press Escape to exit the menu, press Tab to select the option and exit the menu.
            </span>
          </span>
          <div
            class="required-selector__control required-selector__control--is-focused required-selector__control--menu-is-open css-1p30jne-control"
          >
            <div
              class="required-selector__value-container required-selector__value-container--is-multi required-selector__value-container--has-value css-g1d714-ValueContainer"
            >
              <div
                class="css-1rhbuit-multiValue required-selector__multi-value"
              >
                <div
                  class="css-12jo7m5 required-selector__multi-value__label"
                >
                  w
                </div>
                <div
                  class="css-xb97g8 required-selector__multi-value__remove"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="required-selector__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-10-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="required-selector__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <div
                aria-hidden="true"
                class="required-selector__indicator required-selector__clear-indicator css-1gtu0rj-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
              <span
                class="required-selector__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="required-selector__indicator required-selector__dropdown-indicator css-1gtu0rj-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="required-selector__menu css-10caxbu-menu"
          >
            <div
              class="required-selector__menu-list required-selector__menu-list--is-multi css-4ljt47-MenuList"
            >
              <div
                class="required-selector__option required-selector__option--is-disabled required-selector__option--is-focused css-1bw8sh3-option"
                id="react-select-10-option-0"
                tabindex="-1"
              >
                u
              </div>
              <div
                class="required-selector__option required-selector__option--is-disabled css-15iaxgh-option"
                id="react-select-10-option-1"
                tabindex="-1"
              >
                v
              </div>
              <div
                class="required-selector__option css-ofh72q-option"
                id="react-select-10-option-3"
                tabindex="-1"
              >
                x
              </div>
              <div
                class="required-selector__option css-ofh72q-option"
                id="react-select-10-option-4"
                tabindex="-1"
              >
                y
              </div>
              <div
                class="required-selector__option css-ofh72q-option"
                id="react-select-10-option-5"
                tabindex="-1"
              >
                z
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="required-list"
      />
    </div>
    <div>
      <div
        data-testid="restricted-drop-down"
      >
        <div
          class="restricted-selector css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="restricted-selector__control css-1kvltmj-control"
          >
            <div
              class="restricted-selector__value-container restricted-selector__value-container--is-multi restricted-selector__value-container--has-value css-g1d714-ValueContainer"
            >
              <div
                class="css-1rhbuit-multiValue restricted-selector__multi-value"
              >
                <div
                  class="css-12jo7m5 restricted-selector__multi-value__label"
                >
                  u
                </div>
                <div
                  class="css-xb97g8 restricted-selector__multi-value__remove"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="css-1rhbuit-multiValue restricted-selector__multi-value"
              >
                <div
                  class="css-12jo7m5 restricted-selector__multi-value__label"
                >
                  v
                </div>
                <div
                  class="css-xb97g8 restricted-selector__multi-value__remove"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="14"
                    viewBox="0 0 20 20"
                    width="14"
                  >
                    <path
                      d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="restricted-selector__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-11-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="restricted-selector__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <div
                aria-hidden="true"
                class="restricted-selector__indicator restricted-selector__clear-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                  />
                </svg>
              </div>
              <span
                class="restricted-selector__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="restricted-selector__indicator restricted-selector__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="restricted-list"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render nothing when draft has not started and no characters are loaded 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-testid="required-drop-down"
      />
      <div
        data-testid="required-list"
      />
    </div>
    <div>
      <div
        data-testid="restricted-drop-down"
      />
      <div
        data-testid="restricted-list"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render unit dropdowns when draft has not started and characters are loaded 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-testid="required-drop-down"
      >
        <div
          class="required-selector css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="required-selector__control css-1kvltmj-control"
          >
            <div
              class="required-selector__value-container required-selector__value-container--is-multi css-g1d714-ValueContainer"
            >
              <div
                class="required-selector__placeholder css-1wa3eu0-placeholder"
              >
                Required Units
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="required-selector__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-2-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="required-selector__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="required-selector__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="required-selector__indicator required-selector__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="required-list"
      />
    </div>
    <div>
      <div
        data-testid="restricted-drop-down"
      >
        <div
          class="restricted-selector css-2b097c-container"
        >
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="restricted-selector__control css-1kvltmj-control"
          >
            <div
              class="restricted-selector__value-container restricted-selector__value-container--is-multi css-g1d714-ValueContainer"
            >
              <div
                class="restricted-selector__placeholder css-1wa3eu0-placeholder"
              >
                Restricted Units
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class="restricted-selector__input"
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-3-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class="restricted-selector__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="restricted-selector__indicator-separator css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="restricted-selector__indicator restricted-selector__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="restricted-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
